common/lib/xmodule/xmodule/modulestore/perf_tests/generate_report.py

```diff
@@ -0,0 +1,189 @@
+
+"""
+Reads the data generated by performance tests and generates a savable
+report which can be viewed over time to examine the performance effects of code changes on
+various parts of the system.
+"""
+
+import sqlite3
+import numpy as np
+import matplotlib.pyplot as plt
+from lxml.builder import E
+import lxml.html
+import click
+
+
+DB_NAME = 'block_times.db'
+
+
+class HTMLTable(object):
+    """
+    Simple wrapper for an HTML table.
+    """
+    def __init__(self, hdr_columns):
+        self.table = E.TABLE()
+        col_headers = [E.TH(x) for x in hdr_columns]
+        header_row = E.TR(*col_headers)
+        self.table.append(header_row)
+
+    def addRow(self, items):
+        row_items = [E.TD(x) for x in items]
+        self.table.append(E.TR(*row_items))
+
+    def tostring(self):
+        return lxml.html.tostring(self.table)
+
+    @staticmethod
+    def style():
+        return E.style("""
+        table, th, td {
+            border: 1px solid black;
+            border-collapse: collapse;
+        }
+        th, td {
+            padding: 5px;
+        }
+        """
+        )
+
+
+class HTMLDocument(object):
+    """
+    Simple wrapper for an entire HTML document.
+    """
+    def __init__(self, title):
+        self.html = E.html(E.head(E.title("Results"), HTMLTable.style()))
+        self.body = E.body()
+        self.html.append(self.body)
+
+    def addHeader(self, level, text):
+        func_name = "H{}".format(level)
+        self.body.append(getattr(E, func_name)(text))
+
+    def addToBody(self, elem):
+        self.body.append(elem)
+
+    def tostring(self, pretty_print=False):
+        return lxml.etree.tostring(self.html, pretty_print=pretty_print)
```

Comment:
Not sure if it's worth doing at this point, but I probably would have templated this with `mako`.